### PR TITLE
PKG Add pygame-ce

### PR DESCRIPTION
**Pull request by ryanking13**
### Description

Resolve: #289 

Adds `pygame-ce` 2.4.1.

Demo: https://ryanking13.github.io/pyodide-pygame-demo/

__Known Issues__

- freetype doesn't work very well
- some audio related APIs are not supported ([Emscripten does not implement some APIs](https://github.com/emscripten-core/emscripten/blob/2ba2078b00b2a347a59f92256bdc1a42993017f4/src/library_sdl.js#L3548-L3565))
- sdl2_gfx functions doesn't work very well

__Porting existing pygame-based codes__

Following changes are required to run the existing pygame code in the Pyodide.

1. set the canvas to draw on

This is already [described in the docs](https://pyodide.org/en/stable/usage/sdl.html#setting-canvas).

2. yield inside the infinite loop

Usually, pygame codes looks like this:

```python
def run_game():
    while True:
        do_something()
        draw()
        pygame.timer.delay(...)
```

In Pyodide, this needs to be changed to the following:

```python
import asyncio
async def run_game():
    while True:
        do_something()
        draw()
        await asyncio.sleep(...)
```

so that the browser can take control and draw animations on the canvas.

### Checklists

- [x] Add a [CHANGELOG](https://github.com/pyodide/pyodide/blob/main/docs/project/changelog.md) entry
- [x] Add / update tests
- [x] Add new / update outdated documentation

### CODE REVIEW

File: Makefile.envs [hunk 1]
```diff
@@ -149,6 +149,7 @@ export MAIN_MODULE_LDFLAGS= $(LDFLAGS_BASE) \
 	-legl.js \
 	-lwebgl.js \
 	-lhtml5_webgl.js \
+	-lsdl.js \
```
Comment (ryanking13) [inline Makefile.envs:152]: The change in file size (pyodide.asm.js) due to this addition is as follows:

__main branch__

```
    Original size: 1112 KB
    Gzip compressed size (level 1): 248 KB
    Gzip compressed size (level 6): 203 KB
    Gzip compressed size (level 9): 203 KB
    Brotli compressed size: 170 KB
```

__This PR__

```
    Original size: 1194 KB
    Gzip compressed size (level 1): 272 KB
    Gzip compressed size (level 6): 223 KB
    Gzip compressed size (level 9): 223 KB
    Brotli compressed size: 187 KB
```

File: packages/pygame-ce/patches/0001-Don-t-set-font_initialized.patch [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+--- a/src_c/font.c
++++ b/src_c/font.c
+@@ -66,9 +66,7 @@ static unsigned int current_ttf_generation = 0;
+     (((PyFontObject *)(x))->ttf_init_generation == current_ttf_generation)
+ 
+ #if defined(BUILD_STATIC)
+-// SDL_Init + TTF_Init()  are made in main before CPython process the module
+-// inittab so the emscripten handler knows it will use SDL2 next cycle.
+-static int font_initialized = 1;
++static int font_initialized = 0;
```
[the remark this answers is not in the document]
Comment (pmp-p) [reply packages/pygame-ce/patches/0001-Don-t-set-font_initialized.patch:21]: i'm making https://github.com/pygame-community/pygame-ce/pull/2748 so both pygame-web and pyodide do not require patching

[the remark this answers is not in the document]
Comment (ryanking13) [reply packages/pygame-ce/patches/0001-Don-t-set-font_initialized.patch:21]: Thank you so much! I don't know anything about pygame internals, so I really appreciate the various things you do for WASM support.

[the remark this answers is not in the document]
Comment (hoodmane) [reply packages/pygame-ce/patches/0001-Don-t-set-font_initialized.patch:21]: Agreed, we appreciate all the work you've done on this @pmp-p -- it seems like quite a lot of effort.

**Review by hoodmane — APPROVED** [review]
Thanks @ryanking13 and @pmp-p! Looks good to me.

**Comment by pmp-p** [comment]
note that pygame-ce/Panda3D can also render directly in Terminal( like xterm.js with IIP iterm2 image protocol ) with no canvas requirements at all
shot-2024-03-29_1711670160

**Pull request merged**